lib/Controller/SAMLController.php

```diff
@@ -151,9 +151,18 @@ public function login(int $idp = 1): Http\RedirectResponse {
 		switch ($type) {
 			case 'saml':
 				$auth = new Auth($this->samlSettings->getOneLoginSettingsArray($idp));
+				$passthroughParams = $this->config->getSystemValue('user_saml.passthrough_parameters', []);
```

Comment:
```suggestion
				$passthroughParams = $this->config->getAppValue($this->appName, 'passthrough_parameters', []);
```

This should be located in the app config, at least. For this feels very much IdP-config related, I would even say it should be part of `OCA\User_SAML\SAMLSettings`. And there the type can also be enforced, so that `$passthroughParams` is warranted to be an array.

Reply:
Ok, I can see if I can add it to the gui settings then. Stay tuned :).